library/core/src/option.rs

```diff
@@ -1260,6 +1260,19 @@ impl<T> Default for Option<T> {
     }
 }
 
+#[stable(feature = "option_add", since = "1.46.0")]
+impl<T: Add<Output = T>> Add for Option<T> {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self::Output {
+        match (self, other) {
+            (a, None) => a,
+            (None, b) => b,
```

Comment:
Potential footgun, it may not be obvious that `None` is ignored, I thought `None` should become `None` at first if either one is `None`.